src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CKeyStore.java

```diff
@@ -746,9 +755,13 @@ public void engineStore(OutputStream stream, char[] password)
      */
     public void engineLoad(InputStream stream, char[] password)
             throws IOException, NoSuchAlgorithmException, CertificateException {
-        if (stream != null && !keyStoreCompatibilityMode) {
+        if (stream != null && !getRequiresKeyStoreFromFile() && !keyStoreCompatibilityMode) {
             throw new IOException("Keystore input stream must be null");
         }
+        else if (stream == null && getRequiresKeyStoreFromFile())
+        {
+            throw new IOException("Keystore input stream cannot be null");
```

Comment:
Should that message also allude to the fact we are dealing with the keystore from a file use case?

Reply:
The implementation doesn't dictate that the keystore has to come from a file, it could for example be streamed over a socket.  But in all cases an input stream is required if the keystore demands it (in the case of Windows-SST)

Reply:
I'll rename the method and variables to make that clear ie. replace ...fromFile with ...fromStream